src/mlpack/methods/preprocess/preprocess_imputer_main.cpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file preprocess_imputer_main.cpp
+ * @author Keon Kim
+ *
+ * a utility that provides imputation strategies fore
+ * missing values.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/imputer.hpp>
+#include <mlpack/core/data/dataset_info.hpp>
+#include <mlpack/core/data/map_policies/increment_policy.hpp>
+#include <mlpack/core/data/map_policies/missing_policy.hpp>
+#include <mlpack/core/data/imputation_methods/mean_imputation.hpp>
+#include <mlpack/core/data/imputation_methods/median_imputation.hpp>
+#include <mlpack/core/data/imputation_methods/custom_imputation.hpp>
+#include <mlpack/core/data/imputation_methods/listwise_deletion.hpp>
+
+PROGRAM_INFO("Impute Data", "This utility takes a dataset and converts user "
+    "defined missing variable to another to provide more meaningful analysis "
+    "\n\n"
+    "The program does not modify the original file, but instead makes a "
+    "separate file to save the output data; The program requires you to "
+    "specify the file name with --output_file (-o)."
+    "\n\n"
+    "For example, if we consider 'NULL' in dimension 0 to be a missing "
+    "variable and want to delete whole row containing the NULL in the "
+    "column-wise dataset, and save the result to result.csv, we could run"
+    "\n\n"
+    "$ mlpack_preprocess_imputer -i dataset.csv -o result.csv -m NULL -d 0 \n"
+    "> -s listwise_deletion")
+
+PARAM_STRING_REQ("input_file", "File containing data,", "i");
+PARAM_STRING("output_file", "File to save output", "o", "");
+PARAM_STRING("missing_value", "User defined missing value", "m", "")
+PARAM_STRING("strategy", "imputation strategy to be applied", "s", "")
+PARAM_DOUBLE("custom_value", "user_defined custom value", "c", 0.0)
+PARAM_INT("dimension", "the dimension to apply imputation", "d", 0);
+
+using namespace mlpack;
+using namespace arma;
+using namespace std;
+using namespace data;
+
+int main(int argc, char** argv)
+{
+  // Parse command line options.
+  CLI::ParseCommandLine(argc, argv);
+
+  const string inputFile = CLI::GetParam<string>("input_file");
+  const string outputFile = CLI::GetParam<string>("output_file");
+  const string missingValue = CLI::GetParam<string>("missing_value");
+  const double customValue = CLI::GetParam<double>("custom_value");
+  const size_t dimension = (size_t) CLI::GetParam<int>("dimension");
+  string strategy = CLI::GetParam<string>("strategy");
+
+  // missing value should be specified
```

Comment:
First picky comment. Comments should be complete sentences. Take a look at: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#comments